### ISSUE TITLE
TimeLine component for time ranges - Filter does not work correctly 

### ISSUE THREAD

**Issue by ale-cristofori** [geosolutions-it/MapStore2#9268]
## Description
The 'show only times available in the current viewport' filter for the timeline component does not seem to be working correctly. The functionality does not work when no features are present in the viewport (no bars should be shown in the timeline component) and for other types of data the display of the bars seems to be inconsistent with the number of features in the viewport.

## How to reproduce
- Open [this map](https://dev-mapstore.geosolutionsgroup.com/mapstore/#/viewer/44392) 
- There are two time range layers (test_time and donut_2011)
- Zoom to an area where you expect to see no features at all (in the middle of the pacific ocean)
- Activate the filter with the `show only times available in the current viewport` 
- **For the layer donut_2011 only** the filter function won't work even in areas where you expect to see only one feature, some features are removed/filtered but most of them aren't. The filter is inconsistent. 

*Expected Result*
**For the layer test_time** The bars in the timeline histograms should disappear since there should be no feature visible in that remote area. **For the layer donut_2011 only** when only one feature is displayed in the map viewport only one bar should be displayed in the histogram.

*Current Result*
**For the layer test_time**  The bars in the timeline histograms show as if they were not filtered even though no features are visible in that remote area. **For the layer donut_2011 only** when only one feature is displayed in the map viewport many more bars are displayed in the histogram.

- [x] Not browser related

<details><summary> <b>Browser info</b> </summary>

(use this site: <a href="https://www.whatsmybrowser.org/">https://www.whatsmybrowser.org/</a> for non expert users)

| Browser Affected | Version |
|---|---|
|Internet Explorer| |
|Edge| |
|Chrome| |
|Firefox| |
|Safari| |
</details>

## Other useful information
https://github.com/geosolutions-it/MapStore2/assets/14953970/16d59f23-856b-4971-be21-c9d6b70cc819

**Comment by ale-cristofori** [comment]
These issues are part of the Timeline bugs resolutions development, part of
https://github.com/geosolutions-it/support/issues/2428 (original support issue)
https://github.com/geosolutions-it/MapStore2/issues/9208 (MapStore issue)

**Comment by offtherailz** [comment]
In particular I found many problems that made my investigation a little complicated.

1. First the problem seemed to be related to the `bbox` returned by openlayers for the given [given map](https://dev-mapstore.geosolutionsgroup.com/mapstore/#/viewer/44392), at least locally. 
In fact when panning and sooming the bbox on the state of the map, even if I zoom the full world extent, is always something like: 


```
{
  bounds: {
    minx: 741527152488.8824,
    miny: -775530.3289157422,
    maxx: 741547092157.829,
    maxy: 12256677.245593667
  }
}
```

With minx values and miny values very near. This is returned directly by this function here:

https://github.com/geosolutions-it/MapStore2/blob/2e09d7f75e29a94a349408dadad58336659752d6/web/client/components/map/openlayers/Map.jsx#L469

The data returned makes non-sense. Probably related to this issue already notified to openlayers team

https://github.com/openlayers/openlayers/issues/14218

Other maps like the meteorite landings one, do not have this issue.

So I tried to add the donuts layer to the meteorite landings map, and now the bbox is ok, the requests are performed correctly, but the timeline do not update correctly. 

2. So in the conditions where the bbox is correct, in particular, I see the state is updated correctly, but the timeline do not seems to update.

**This part of the  investigation is still in progress...  I will update this point as well as I have finished it**


3. First of all I found another bug I notify. Not strictly related to this issue. This happens in the case the timeline is closed but the filter by viewport is active. When panning:

https://github.com/geosolutions-it/MapStore2/blob/2e09d7f75e29a94a349408dadad58336659752d6/web/client/epics/timeline.js#L128

This line doesn't take into account the case when the domain returns an array. In this case I have an error in console.
```
EpicsUtils.js:28 Error in epic "updateRangeDataOnRangeChange". Original error: TypeError: Cannot read properties of null (reading '0')
    at roundResolution (TimeUtils.js:193:65)
    at roundRangeResolution (TimeUtils.js:212:22)
    at loadRangeData (timeline.js:133:78)
    at eval (timeline.js:431:95)
    at Array.map (<anonymous>)
    at SwitchMapSubscriber.eval [as project] (timeline.js:431:85)
    at SwitchMapSubscriber._next (switchMap.js:88:27)
    at Subscriber.next (Subscriber.js:89:18)
    at OuterSubscriber.notifyNext (OuterSubscriber.js:19:26)
    at InnerSubscriber._next (InnerSubscriber.js:23:21)
    at Subscriber.next (Subscriber.js:89:18)
    at DebounceTimeSubscriber.debouncedNext (debounceTime.js:98:30)
    at AsyncAction.dispatchNext (debounceTime.js:114:16)
    at AsyncAction._execute (AsyncAction.js:111:18)
    at AsyncAction.execute (AsyncAction.js:86:26)
    at AsyncScheduler.flush (AsyncScheduler.js:36:32)
overrideMethod @ console.js:213
eval @ EpicsUtils.js:28
CatchSubscriber.error @ catch.js:104
Subscriber._error @ Subscriber.js:128
Subscriber.error @ Subscriber.js:102
Subscriber._error @ Subscriber.js:128
Subscriber.error @ Subscriber.js:102
Subscriber._error @ Subscriber.js:128
Subscriber.error @ Subscriber.js:102
SwitchMapSubscriber._next @ switchMap.js:91
Subscriber.next @ Subscriber.js:89
OuterSubscriber.notifyNext @ OuterSubscriber.js:19
InnerSubscriber._next @ InnerSubscriber.js:23
Subscriber.next @ Subscriber.js:89
DebounceTimeSubscriber.debouncedNext @ debounceTime.js:98
dispatchNext @ debounceTime.js:114
AsyncAction._execute @ AsyncAction.js:111
AsyncAction.execute @ AsyncAction.js:86
AsyncScheduler.flush @ AsyncScheduler.js:36
setInterval (async)
AsyncAction.requestAsyncId @ AsyncAction.js:65
AsyncAction.schedule @ AsyncAction.js:60
Scheduler.schedule @ Scheduler.js:43
DebounceTimeSubscriber._next @ debounceTime.js:89
Subscriber.next @ Subscriber.js:89
OuterSubscriber.notifyNext @ OuterSubscriber.js:19
InnerSubscriber._next @ InnerSubscriber.js:23
Subscriber.next @ Subscriber.js:89
FilterSubscriber._next @ filter.js:88
Subscriber.next @ Subscriber.js:89
MapSubscriber._next @ map.js:83
Subscriber.next @ Subscriber.js:89
FilterSubscriber._next @ filter.js:88
Subscriber.next @ Subscriber.js:89
WithLatestFromSubscriber._next @ withLatestFrom.js:113
Subscriber.next @ Subscriber.js:89
Subject.next @ Subject.js:55
eval @ createEpicMiddleware.js:80
eval @ createEpicMiddleware.js:79
eval @ redux-logger.js:1
eval @ index.js:12
eval @ bindActionCreators.js:7
onClick @ Timeline.jsx:394
onClick @ buttonWithDisabled.jsx:32
callCallback @ react-dom.development.js:362
invokeGuardedCallbackDev @ react-dom.development.js:411
invokeGuardedCallback @ react-dom.development.js:466
invokeGuardedCallbackAndCatchFirstError @ react-dom.development.js:480
executeDispatch @ react-dom.development.js:612
executeDispatchesInOrder @ react-dom.development.js:637
executeDispatchesAndRelease @ react-dom.development.js:743
executeDispatchesAndReleaseTopLevel @ react-dom.development.js:752
forEachAccumulated @ react-dom.development.js:724
runEventsInBatch @ react-dom.development.js:769
runExtractedPluginEventsInBatch @ react-dom.development.js:914
handleTopLevel @ react-dom.development.js:5848
batchedEventUpdates$1 @ react-dom.development.js:24343
batchedEventUpdates @ react-dom.development.js:1463
dispatchEventForPluginEventSystem @ react-dom.development.js:5943
attemptToDispatchEvent @ react-dom.development.js:6059
dispatchEvent @ react-dom.development.js:5963
unstable_runWithPriority @ scheduler.development.js:815
runWithPriority$2 @ react-dom.development.js:12188
discreteUpdates$1 @ react-dom.development.js:24359
discreteUpdates @ react-dom.development.js:1486
dispatchDiscreteEvent @ react-dom.development.js:5926
Show 1 more frame
Show less
EpicsUtils.js:31 Uncaught TypeError: Cannot read properties of null (reading '0')
    at roundResolution (TimeUtils.js:193:65)
    at roundRangeResolution (TimeUtils.js:212:22)
    at loadRangeData (timeline.js:133:78)
    at eval (timeline.js:431:95)
    at Array.map (<anonymous>)
    at SwitchMapSubscriber.eval [as project] (timeline.js:431:85)
    at SwitchMapSubscriber._next (switchMap.js:88:27)
    at Subscriber.next (Subscriber.js:89:18)
    at OuterSubscriber.notifyNext (OuterSubscriber.js:19:26)
    at InnerSubscriber._next (InnerSubscriber.js:23:21)
    at Subscriber.next (Subscriber.js:89:18)
    at DebounceTimeSubscriber.debouncedNext (debounceTime.js:98:30)
    at AsyncAction.dispatchNext (debounceTime.js:114:16)
    at AsyncAction._execute (AsyncAction.js:111:18)
    at AsyncAction.execute (AsyncAction.js:86:26)
    at AsyncScheduler.flush (AsyncScheduler.js:36:32)
```
anyway this is not properly related to this case.

**Comment by ale-cristofori** [comment]
@offtherailz, the client agreed to use two more hours of investigation as asked.
We have 2 extra more hours to conduct the investigation

**Comment by ale-cristofori** [comment]
Hello @offtherailz, I am testing with the `donut 2011`. And comparing it with dev (that does not have the fix). I built the war and running a local instance of Tomcat connecting to GS-Stable.
I can see the bars representing the intervals appearing and disappearing and an infinite loop error in the JS console  see below. The graphics behaves buggy.  See below


https://github.com/geosolutions-it/MapStore2/assets/14953970/2903d22f-bf9c-47d1-8d05-cfad7569135b

**Comment by offtherailz** [comment]
I comment just for making note that the issue was probably due to a wrong map center calculation that makes OL generate wrong bboxes for filtering. 
Anyway, it is not clear the cause that generated the map reported. I attach the map here:


[map_15_vlanderen.json](https://github.com/geosolutions-it/MapStore2/files/13486902/map_15_vlanderen.json)

the issue is cause by the `center.x` 

```json
{"version":2,"map":{"center":{"x":6662525.329597231,"y":50.652011207093054,"crs":"EPSG:4326"} ...}
```

**Comment by ale-cristofori** [comment]
@tdipisa, this issue needs to be re-opened to allow further testing on the latest issue found by the client.
If a new issue to cover this newly found issue is required, let me know.

**Comment by tdipisa** [comment]
@ale-cristofori please open a new issue reporting the steps to reproduce etc as usual. Thank you.

**Comment by ale-cristofori** [comment]
closed in favour of #9843
